domain/storageprovisioning/state/filesystem.go

```
@@ -587,3 +587,137 @@ WHERE  provision_scope_id=0
+// filesystem ID.
+//
+// The following errors may be returned:
+// - [storageprovisioningerrors.FilesystemNotFound] when no filesystem
+// exists for the provided filesystem id.
+func (st *State) GetFilesystemUUIDByID(
+	ctx context.Context, filesystemIDStr string,
+) (string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	fsID := filesystemID{ID: filesystemIDStr}
+	stmt, err := st.Prepare(`
+SELECT &entityUUID.*
+FROM   storage_filesystem
+WHERE  filesystem_id = $filesystemID.filesystem_id`, fsID, entityUUID{})
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	fsUUID := entityUUID{}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, stmt, fsID).Get(&fsUUID)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("filesystem %q not found", filesystemIDStr).
+				Add(storageprovisioningerrors.FilesystemNotFound)
+		} else if err != nil {
+			return errors.Capture(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+	return fsUUID.UUID, nil
+}
+
+// SetFilesystemProvisionedInfo sets on the provided filesystem the
+// information about the provisioned filesystem.
+func (st *State) SetFilesystemProvisionedInfo(
+	ctx context.Context,
+	filesystemUUID string,
+	info storageprovisioning.FilesystemProvisionedInfo,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	fs := filesystemProvisionedInfo{
+		UUID:       filesystemUUID,
+		ProviderID: info.ProviderID,
+		SizeMiB:    info.SizeMiB,
+	}
+	stmt, err := st.Prepare(`
+UPDATE storage_filesystem
+SET    provider_id = $filesystemProvisionedInfo.provider_id,
+       size_mib = $filesystemProvisionedInfo.size_mib
+WHERE  uuid = $filesystemProvisionedInfo.uuid
+`, fs)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, fs).Run()
```

Do you want to raise a not found error here as well?

---

We should. The life PR has this.

```
 checkFilesystemExists checks if a filesystem for the provided uuid exists.
// True is returned when the filesystem exists.
func (st *State) checkFilesystemExists(
	ctx context.Context,
	tx *sqlair.TX,
	uuid domainstorageprovisioning.FilesystemUUID,
) (bool, error) {
	entityUUIDInput := entityUUID{UUID: uuid.String()}

	checkQuery, err := st.Prepare(`
SELECT &entityUUID.*
FROM   storage_filesystem
WHERE  uuid = $entityUUID.uuid
`,
		entityUUIDInput,
	)
	if err != nil {
		return false, errors.Capture(err)
	}

	err = tx.Query(ctx, checkQuery, entityUUIDInput).Get(&entityUUIDInput)
	if errors.Is(err, sqlair.ErrNoRows) {
		return false, nil
	} else if err != nil {
		return false, errors.Capture(err)
	}

	return true, nil
}
```